### PR TITLE
Duplo-18045 TF: GCP: Add Terraform support to create bucket with multi region and region

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Implemented `duplocloud_gcp_s3_bucket` resource with multi region support
## Summary of changes

This PR does the following:

- Implemented duplocloud_gcp_s3_bucket resource operations
- Updated docs with multi region example

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

File: duplocloud/resource_duplo_s3_bucket.go [hunk 1]
```diff
@@ -425,6 +425,7 @@ func resourceS3BucketSetData(d *schema.ResourceData, tenantID string, name strin
 	d.Set("managed_policies", duplo.Policies)
 	d.Set("tags", keyValueToState("tags", duplo.Tags))
 	d.Set("region", duplo.Region)
+	d.Set("location", duplo.Location)
```
Comment (komalkadam-dc) [inline duplocloud/resource_duplo_s3_bucket.go:428]: I don't see this attribute supported for S3 bucket.

**Review by komalkadam-dc — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_s3_bucket.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_s3_bucket.go:428]: Screenshot 2024-06-27 at 20 47 25
It was in qa-gcp its been changed to different release, but found in qa-gcp2. Check attachment

Comment (komalkadam-dc) [reply duplocloud/resource_duplo_s3_bucket.go:428]: isn't it AWS s3 bucket?
I was saying this attribute is not present for AWS S3 bucket, its present gcp

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_s3_bucket.go:428]: Its gcp specific

Comment (tahir-duplo) [reply duplocloud/resource_duplo_s3_bucket.go:428]: @Duplo-nikv Let remove location from `resourceS3BucketSetData` method and new gcp specific method say `resourceS3GcpBucketSetData`

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_s3_bucket.go]: done changes

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Review by komalkadam-dc — APPROVED** [review]

**Pull request merged**